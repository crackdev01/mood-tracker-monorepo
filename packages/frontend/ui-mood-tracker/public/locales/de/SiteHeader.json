{
  "header": "Mood Tracker",
  "moodEntry": "Stimmungseintrag",
  "statistics": "Statistiken",
  "menu": {
    "actions": {
      "location": "Current position",
      "changeLanguage": "Switch to English",
      "logout": "Logout"
    }
  }
}
