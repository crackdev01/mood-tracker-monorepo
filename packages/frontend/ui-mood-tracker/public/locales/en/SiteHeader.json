{
  "header": "Mood Tracker",
  "moodEntry": "Mood Entry",
  "statistics": "Statistics",
  "menu": {
    "actions": {
      "location": "Current position",
      "changeLanguage": "Switch to Deutsche",
      "logout": "Logout"
    }
  }
}
