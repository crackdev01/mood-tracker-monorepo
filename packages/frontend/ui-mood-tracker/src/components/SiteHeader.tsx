import React from 'react';

import '../App.scss';

function SiteHeader() {
  return <h2>Mood Tracker</h2>;
}

export default SiteHeader;
