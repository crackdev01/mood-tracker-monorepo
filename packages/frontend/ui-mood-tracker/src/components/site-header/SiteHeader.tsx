import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { Link, useLocation } from 'react-router-dom';
import { Dropdown, Header, Menu } from 'semantic-ui-react';

import { UserActions } from '../../store/user/types';
import { ApplicationState } from '../../store';

import './site-header.scss';

enum LocaleEnum {
  English = 'en',
  Deutsche = 'de',
}

const SiteHeader = () => {
  const dispatch = useDispatch();
  const { t, i18n } = useTranslation(['SiteHeader']);
  const user = useSelector((state: ApplicationState) => state.userReducer.user);
  const [currentUser, setCurrentUser] = useState('');
  const location = useLocation();

  const isAuthenticated = user ? !!user.accessToken : false;

  const updateLanguage = () => {
    const { language } = i18n;
    i18n.changeLanguage(language === LocaleEnum.English ? LocaleEnum.Deutsche : LocaleEnum.English);
  };

  const logout = () => {
    dispatch({
      type: UserActions.LOGOUT_USER,
    });
  };

  useEffect(() => {
    const username = user.decodedAccessToken ? user.decodedAccessToken.username : '';
    setCurrentUser(username);
  }, [user]);

  return (
    <Menu pointing secondary className="site-header">
      <Menu.Item header>
        <Header as="h3">{t('header')}</Header>
      </Menu.Item>
      {isAuthenticated && (
        <Menu.Item active={location.pathname === '/mood-entry' || location.pathname === '/'}>
          <Link to="/mood-entry">{t('moodEntry')}</Link>
        </Menu.Item>
      )}
      {isAuthenticated && (
        <Menu.Item active={location.pathname === '/statistics'}>
          <Link id="statistics" to="/statistics">
            {t('statistics')}
          </Link>
        </Menu.Item>
      )}
      {isAuthenticated && (
        <Menu.Menu position="right">
          <Dropdown id="toggle-menu" item text={currentUser}>
            <Dropdown.Menu>
              <Dropdown.Item>{t('menu.actions.location')}</Dropdown.Item>
              <Dropdown.Item id="change-language" onClick={updateLanguage}>
                {t('menu.actions.changeLanguage')}
              </Dropdown.Item>
              <Dropdown.Item id="logout" onClick={logout}>
                {t('menu.actions.logout')}
              </Dropdown.Item>
            </Dropdown.Menu>
          </Dropdown>
        </Menu.Menu>
      )}
    </Menu>
  );
};

export default SiteHeader;
