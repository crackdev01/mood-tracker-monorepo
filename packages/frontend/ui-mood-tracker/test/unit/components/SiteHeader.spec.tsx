import React from 'react';
import { render } from '@testing-library/react';

import SiteHeader from '../../../src/components/SiteHeader';

describe('SiteHeader', () => {
  test('renders', () => {
    const { container } = render(<SiteHeader />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
