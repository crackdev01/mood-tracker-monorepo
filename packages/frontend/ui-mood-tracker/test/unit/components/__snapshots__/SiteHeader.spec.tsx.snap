// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader renders 1`] = `
<h2>
  Mood Tracker
</h2>
`;
